packages/nx/src/executors/run-commands/running-tasks.ts

```diff
@@ -122,33 +122,59 @@ export class ParallelRunningTasks implements RunningTask {
         cb(code, terminalOutput);
       }
     } else {
-      const results = await Promise.all(
+      const runningProcesses = new Set<RunningNodeProcess>();
+      let hasFailure = false;
+      let failureDetails: {
+        childProcess: RunningNodeProcess;
+        code: number;
+        terminalOutput: string;
+      } | null = null;
+      const terminalOutputs = new Map<RunningNodeProcess, string>();
+
+      await Promise.allSettled(
         this.childProcesses.map(async (childProcess) => {
+          runningProcesses.add(childProcess);
+
           childProcess.onOutput((terminalOutput) => {
             for (const cb of this.outputCallbacks) {
               cb(terminalOutput);
             }
           });
-          const result = await childProcess.getResults();
```

Comment:
`getResults` already waits for process exit..

Don't we just need to check the result to see if other processes should be terminated?